### ISSUE TITLE
Use Proxy When Fetching Server List

### ISSUE THREAD

**Issue by Riajyuu** [bitbeans/SimpleDnsCrypt#496]
Now raw.githubcontent.com is completely blocked in China. Whenever users click server tabs, the GUI crash instantly. This means some solutions:

1. simply don't quit
2. respect system proxy when updating the list
3. ask to set a proxy, like SOCKS5 before updating

**Comment by LittleGrocery** [comment]
I agree with you. The software crash multiple times in my machine. What's more, Google doh server can not access in China, so does cloudflare. Could developer add a proxy function like socks5 to support the doh server access which banned in China?
